components/diagnosis-suggestion/api/src/main/java/org/phenotips/diagnosis/internal/DefaultDiagnosisService.java

```diff
@@ -202,11 +208,17 @@ public int compare(Integer o1, Integer o2)
         return results;
     }
 
-    private void addTermAndAncestors(Term t, Observations o)
+    private Boolean addTermAndAncestors(Term t, Observations o)
```

Comment:
There's a difference between the primitive type `boolean` and the class `Boolean`, and you seem to be using a primitive in the rest of the code. Use `boolean` as the return type.